### DISCUSSION TITLE
Server options, explained

### DISCUSSION THREAD

**Discussion by dkmj** [jamulussoftware/jamulus#1507] [Technical Support & Help] [answered]
Sorry if this is in the docs already... I'm running couple of servers for my choir rehearsal in Covid-times. we have issues, mostly on tech know how and equip on the singer side, but i might also do a better job on the server config setup side.

I run on a MacMini 2012 core17 w 32GB ram and no other significant load. Always lots of memory and processor never seems to go above 20% load. 250MBit symmetrical bandwidth on a home fiber connection.

1. I had -F option set. is that 'bad' for sound processing/quality? Could turning it off 'help', especially in the beginning when we still have a few singers with terrible audio and high latency?
2. -s server option. running a server without it seems to work too? what's the effect if the -s flag? 
3. -T multithreading? valid on all platforms? how does it affect server sound quality, if at all? recommended to turn on or off? 
4. -u i've set my private server limit to 35 which is the upper limit of singers joining at one time. Could I win 'sound quality' by setting this lower, or not specifying at all?
5. -l Running a log might help me (?) understanding what happens but would affect performance negatively, or?

thanks a million, this is fun!
/Peter in Stockholm Sweden

**Comment by gilgongo** [comment, marked as answer]
Hi Peter - in as [per the docs](https://jamulus.io/wiki/Running-a-Server) on this, I'm afraid that the vast majority of issues you will be having, if not all of them in fact, will be client side and not to do with your server setup: 

> Many people attribute problems to the server that are in fact problems with the client. Much depends on the clients’ hardware, the networks that they are on, and whether they are sticking to Rule Number One. 

But to answer your questions:

On 1, it depends :-) If users check the "Enable small network buffers" option in their client settings, then they need a sound card that has a buffer delay smaller than 128 samples. The smaller the network buffers, the lower the audio latency. But at the same time the network load increases and the probability of audio dropouts also increases.

On 2, the -s flag puts Jamulus into server mode. I'm pretty sure you can't run a server without it!

On 3, it sounds like your processor will be fine with it on. If you get issues try turning it off, but I highly doubt it will  do much as your ensemble isn't very large - and you have plenty of CPU.

On 4. I doubt that will help. Your CPU and bandwidth aren't even breaking a sweat. I would set it to a slightly bigger limit in fact just in case somebody wants to come and listen :-)

On 5, there isn't much in the log, but running one won't harm anything.

**Reply by dkmj** [reply]
Thanks, as I suspected then. I did run without the -s server flag and it throws a server GUI nonetheless? We'll as participants to not go too low settings on the latency to limit dropouts. Indeed, many singers are on a steep learning curve, and some equipment purchases are yet to happen.
